### DISCUSSION TITLE
Need help with bridging Micrometer with Otel in reactor-netty

### DISCUSSION THREAD

**Discussion by KaczDev** [open-telemetry/opentelemetry-java-instrumentation#12821] [Q&A]
Hi everyone, 

I'm interested in collecting netty metrics. By default netty provides a Micrometer implementation but it's possible to provide [custom metric registrars](https://projectreactor.io/docs/netty/snapshot/reference/http-client.html#_connection_pool).

I cannot enable micrometer instrumentation as it doubles a lot of my already established metrics.

This is how the interface is defined
```java
    public interface MeterRegistrar {
        void registerMetrics(String var1, String var2, SocketAddress var3, ConnectionPoolMetrics var4);

        default void deRegisterMetrics(String poolName, String id, SocketAddress remoteAddress) {
        }
    }
```

I have created the custom metrics registrar. I used the opentelemetry-micrometer bridge library but my metrics are not exported.
```java
public class OpenTelemetryMetricsRegistrar implements ConnectionProvider.MeterRegistrar
{
    private final MeterRegistry REGISTRY;
    
    public OpenTelemetryMetricsRegistrar()
    {
        this.REGISTRY = OpenTelemetryMeterRegistry.builder(GlobalOpenTelemetry.get())
            .setPrometheusMode(false)
            .build();
    }
    
    @Override
    public void registerMetrics(String poolName, String id, SocketAddress remoteAddress, ConnectionPoolMetrics metrics)
    {
        String addressAsString = formatSocketAddress(remoteAddress);
        Tags tags = Tags.of(ID, id, REMOTE_ADDRESS, addressAsString, NAME, poolName);
        Gauge.builder(CONNECTION_PROVIDER_PREFIX + TOTAL_CONNECTIONS, metrics, ConnectionPoolMetrics::maxAllocatedSize)
            .tags(tags)
            .register(REGISTRY);
    }
    
    @Override
    public void deRegisterMetrics(String poolName, String id, SocketAddress remoteAddress)
    {
        String addressAsString = formatSocketAddress(remoteAddress);
        Tags tags = Tags.of(ID, id, REMOTE_ADDRESS, addressAsString, NAME, poolName);
        REGISTRY.remove(new Meter.Id(CONNECTION_PROVIDER_PREFIX + TOTAL_CONNECTIONS, tags, null, null, Meter.Type.GAUGE));
    }
```
I have tried with `otel.instrumentation.micrometer.enabled` enabled and disabled and I have the same results.
Not sure what should I do to have those metrics exported via OTEL channels.

For now I'm interested only in Connection Pool metrics. I'd be happy to create a PR afterwards to improve the reactor-netty instrumentation with metrics.

I also have netty instrumentations enabled
```
otel.instrumentation.netty.enabled=true
otel.instrumentation.reactor.enabled=true
otel.instrumentation.reactor-netty.enabled=true
```

**Comment by Cirilla-zmh** [comment]
Could you share how do you export metrics by `otel-java-instrumentation`? Or could you give a demo?
I've looked into some mechanisms of micrometer and I'm glad to help you. :)

**Comment by KaczDev** [comment]
I usually rely on Opentelemetry-javaagent being added to the JVM. In the code, when I want to add my own otel metrics, I just call GlobalOpentelemetry. Thought that here `OpenTelemetryMetricsRegistrar` does that for me. Is that right?

**Reply by Cirilla-zmh** [reply]
I'm sorry, but I haven't used micrometer with reactor-netty, and it seems that `MeterRegistrar` is part of its observability extensions. I will attempt to create a demo and verify this.
